core/EAP7-1266_WFCORE-4314_enchance_keystore_commands_alias.adoc

```diff
@@ -0,0 +1,343 @@
+
+=== Nice-to-Have Requirements
+
+
+
+=== Non-Requirements
+
+=== Design
+
+Preparation:
+[source]
+--
+/subsystem=elytron/key-store=serverKS:add(path=server.keystore.jsk,relative-to=jboss.server.config.dir,credential-reference={clear-text=secret}, type=JKS)
+cd subsystem=elytron/key-store=serverKS
+:generate-key-pair(alias=server1,distinguished-name="CN=www.example.com")
+[standalone@localhost:9990 key-store=serverKS] :generate-key-pair(alias=server2,distinguished-name="CN=www.example.com")
+[standalone@localhost:9990 key-store=serverKS] :generate-key-pair(alias=backup,distinguished-name="CN=www.example.com")
+--
+
+
+`read-aliases` - no change in default output.
+[source]
+--
+[standalone@localhost:9990 key-store=serverKS] :read-aliases
+{
+    "outcome" => "success",
+    "result" => [
+        "backup",
+        "server2",
+        "server1"
+    ]
+}
+--
+
+
+`read-aliases` - inclusion of single parameter does not change output.
+[source]
+--
+[standalone@localhost:9990 key-store=serverKS] :read-aliases(verbose=true)
+{
+    "outcome" => "success",
+    "result" => [
+        "backup",
+        "server2",
+        "server1"
+    ]
+}
+--
+
+
+`read-aliases` - recursive read, without hex field included.
+[source]
+--
+[standalone@localhost:9990 key-store=serverKS] :read-aliases(recursive=true)
+{
+    "outcome" => "success",                                                                                                                                                                                                                         
+    "result" => {                                                                                                                                                                                                                                   
+        "backup" => {                                                                                                                                                                                                                               
+            "alias" => "backup",                                                                                                                                                                                                                    
+            "entry-type" => "PrivateKeyEntry",                                                                                                                                                                                                      
+            "creation-date" => "2019-05-27T08:29:42.093+0200",                                                                                                                                                                                      
+            "certificate-chain" => [{
+                "type" => "X.509",
+                "algorithm" => "DSA",
+                "format" => "X.509",
+                "public-key" => undefined,
```

Comment:
Instead of having the `public-key` appear as `undefined` to make the output shorter, I think it would be better just to not include `public-key` in the output at all.

Reply:
done